### PR TITLE
Update stubs with new ujson features

### PR DESCRIPTION
**Pull request by dbergamin**
Reference from ujson 5.7.0:
https://github.com/ultrajson/ultrajson/blob/5.7.0/python/objToJSON.c#L797 https://github.com/ultrajson/ultrajson/blob/5.7.0/tests/test_ujson.py

### CODE REVIEW

File: stubs/ujson/ujson.pyi [hunk 1]
```diff
@@ -10,6 +10,10 @@ def encode(
     escape_forward_slashes: bool = ...,
     sort_keys: bool = ...,
     indent: int = ...,
+    allow_nan: bool = ...,
+    reject_bytes: bool = ...,
+    default: Any = ...,
+    separators: Any = ...,
```
Comment (srittau) [inline stubs/ujson/ujson.pyi:16]:
```suggestion
    default: Callable[[Unknown], Unknown] | None = None,
    separators: tuple[str, str] | None = None,
```

Same for the other two instances below. Needs import of `_typeshed.Unknown` and possible `Callable`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks! A few further type improvements below:

File: stubs/ujson/ujson.pyi (hunk 1 above)
Comment (srittau) [reply stubs/ujson/ujson.pyi:16]: Yikes, indeed.

Comment (dbergamin) [reply stubs/ujson/ujson.pyi]: thanks for the feedback! have improved the precision as suggested

**Pull request merged**